File: dotnet/xmsclient/Client/State/Listener/ConnectionCloseOkListener.cs
using System;
using jpmorgan.mina.common;
using log4net;
using OpenAMQ.Framing;
using OpenAMQ.XMS.Client.Protocol;

namespace OpenAMQ.XMS.Client.State.Listener
{
    public class ConnectionCloseOkListener : BlockingMethodFrameListener
    {

        public ConnectionCloseOkListener()
            : base(0)
        {
        }

        public bool ProcessMethod(int channelId, AMQMethodBody frame)
        {
            return (frame is ConnectionCloseOkBody);
        }
    }
}

File: java/src/org/openamq/client/state/listener/ConnectionCloseOkListener.java
package org.openamq.client.state.listener;

import org.openamq.AMQException;
import org.openamq.client.protocol.BlockingMethodFrameListener;
import org.openamq.framing.AMQMethodBody;
import org.openamq.framing.ConnectionCloseOkBody;

public class ConnectionCloseOkListener extends BlockingMethodFrameListener
{

    public ConnectionCloseOkListener()
    {
        super(0);
    }
    
    public boolean processMethod(int channelId, AMQMethodBody frame) throws AMQException
    {
        return (frame instanceof ConnectionCloseOkBody);
    }
}
